sql/src/main/java/io/crate/operation/collect/sources/InformationSchemaIterables.java

```diff
@@ -99,10 +107,28 @@ public InformationSchemaIterables(final Schemas schemas,
         return sqlFeatures;
     }
 
+    public Iterable<IngestionRuleInfo> ingestionRules() {
+        return ingestionRules;
+    }
+
     public Iterable<Void> keyColumnUsageInfos() { return keyColumnUsages; }
 
     public Iterable<Void> referentialConstraintsInfos() { return referentialConstraints; }
 
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        if (event.metaDataChanged() == false) {
+            return;
+        }
+        createMetaDataBasedIterables(event.state().getMetaData());
+    }
+
+    private void createMetaDataBasedIterables(MetaData metaData) {
+        RoutineInfos routineInfos = new RoutineInfos(fulltextAnalyzerResolver, metaData);
+        routines = FluentIterable.from(routineInfos).filter(Objects::nonNull);
+        IngestionRuleInfos ingestionRuleInfos = new IngestionRuleInfos(metaData);
+        ingestionRules = FluentIterable.from(ingestionRuleInfos).filter(Objects::nonNull);
```

Comment:
seems like the `ingestionRuleInfos` never contain any null entries - shouldn't be necessary to apply a filter